scimono-compliance-tests/src/main/java/com/sap/scimono/scim/system/tests/SCIMComplianceTest.java

```diff
@@ -70,11 +74,29 @@
 
   protected SCIMComplianceTest() {
     SCIMClientService scimClientService = configureScimClientService(SERVICE_URL);
-    userRequest = scimClientService.buildUserRequest();
-    groupRequest = scimClientService.buildGroupRequest();
-    schemaRequest = scimClientService.buildSchemaRequest();
+    SCIMRequest.Builder builder = SCIMRequest.newBuilder();
+    populateCustomHeaders(builder);
+
+    userRequest = scimClientService.buildUserRequest(builder);
+    groupRequest = scimClientService.buildGroupRequest(builder);
+    schemaRequest = scimClientService.buildSchemaRequest(builder);
   }
 
+  public static SCIMRequest.Builder populateCustomHeaders(SCIMRequest.Builder newBuilder) {
```

Comment:
Since this is method returns `SCIMRequest.Builder` it would be better to add more appropriate name: e.g `requestWithCustomHeaders` or 'requestBuilderWithCustomHeaders'